src/lambda_codebase/account_processing/delete_default_vpc.py

```diff
@@ -25,12 +28,25 @@ def assume_role(account_id):
         "adf_delete_default_vpc",
     )
 
-
+@tenacity.retry(
+    retry=tenacity.retry_if_exception_type(ClientError),
+    # Fail after 180 Sec of retrying
+    stop=tenacity.stop_after_delay(180),
+    # Wait 2^x * 1 second between each retry starting with 4s, max 10s intervals
+    wait=tenacity.wait_exponential(multiplier=1, min=4, max=10),
+)
```

Comment:
Is there any reason why Boto's retry mechanism isn't enough here?

Reply:
To be honest, I'm not sure. Would you have an example of how to use that and write tests for it?

I've moved to tenacity because it's used here: https://github.com/awslabs/aws-deployment-framework/blob/5d6e5ce2b4c0d01e670da0e2fd13e80bdab618a6/src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/cloudformation.py#L13